src/TestSuite/IntegrationTestCase.php

```diff
@@ -243,6 +259,17 @@ public function enableCsrfToken()
     }
 
     /**
+     * Calling this method will re-store flash messages into the test session
+     * after being removed by the FlashHelper
+     *
+     * @return void
+     */
+    public function enableRememberFlashMessages()
+    {
+        $this->_rememberFlashMessages = true;
```

Comment:
If people rely on the messages being removed it could break their tests, that's the only reason I made it opt in. A user basically re-testing that the flash component removes the message after render would be the only issue I can think of.